virtwho/virt/rhevm/rhevm.py

```diff
@@ -71,18 +71,73 @@ def __init__(self, logger, config):
         self.username = self.config.username
         self.password = self.config.password
 
-        self.clusters_url = urlparse.urljoin(self.url, "api/clusters")
-        self.hosts_url = urlparse.urljoin(self.url, "api/hosts")
-        self.vms_url = urlparse.urljoin(self.url, "api/vms")
-
         self.auth = HTTPBasicAuth(self.config.username, self.config.password)
 
+    @property
+    def clusters_url(self):
+        if not hasattr(self, '_clusters_url'):
+            if self.major_version == '4' and 'ovirt-engine' not in self.url:
```

Comment:
I was having issues with mocking items by doing the version check on init.  Since the version check was reaching out and doing a request.get, it was creating complicated testing patterns.  I switched to using properties based on khowell's suggestion.

With this modification, I can just set 'self._major_version' on the object in the unit test setup and ensure that we short-circuit the version testing logic for those test.